### DISCUSSION TITLE
[Fast Data] [SVC Low Code] Reference to the same projection

### DISCUSSION THREAD

**Discussion by flower-of-the-bridges** [mia-platform/community#36] [Q&A] [answered]
Hi

I need to link a projection, namely `PROJ_1`, to itself under the following condition: 
```
{
  'FIELD_1': '__string__[value]'
  'FIELD_2': 'FIELD_3' 
}
``` 

This condition, that we can assume is named `CONDITION_1`, is then used in the aggregation in the following way:

```
config: {
    SV_CONFIG: {
      dependencies: {
        PROJ_1: {
          type: 'projection',
          on: '_identifier',
        },
        PROJ_1_DUPLICATE: {
          type: 'projection',
          aliasOf: 'PROJ_1',
          on: 'CONDITION_1',
        }
    },
    mapping: {
    // ...
    mapping1: 'PROJ_1_DUPLICATE.FIELD'
    }
}
```
When the condition is not met, I expect to have `mapping1: null`.
However, the field `mapping1` is not returned by the aggregator.

I'm approaching the problem correctly or there is another solution that fits better to this case?

**Comment by STartarotti** [comment, marked as answer]
Hi,

at the moment this particular case is not among the ones supported by the aggregation schema.
However you could always define a custom function with the [`__fromFile__[fileName]` feature](https://docs.mia-platform.eu/docs/fast_data/configuration/single_view_creator/low_code#mapping:~:text=function%20result%3A%20when%20using%20a%20custom%20function%20to%20compute%20the%20value%20of%20the%20single%20view%20field%2C%20expressed%20with%20this%20syntax%3A%20%22fromFileField%22%3A%20%22__fromFile__%5BfileName%5D%22) of the mapping.
